x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/extractor/DataFrameDataExtractor.java

```diff
@@ -91,9 +92,28 @@ public void cancel() {
 
     protected List<Row> initScroll() throws IOException {
         LOGGER.debug("[{}] Initializing scroll", context.jobId);
-        SearchResponse searchResponse = executeSearchRequest(buildSearchRequest());
-        LOGGER.debug("[{}] Search response was obtained", context.jobId);
-        return processSearchResponse(searchResponse);
+        return tryRequestWithSearchResponse(() -> executeSearchRequest(buildSearchRequest()));
+    }
+
+    private List<Row> tryRequestWithSearchResponse(Supplier<SearchResponse> request) throws IOException {
```

Comment:
I think having this method is a nice way of handling with the 2 different requests in a single place. If people like it, we can reuse this in `ScrollDataExtractor` to simplify and rely on `allow_partial_search_results` to do its job of clearing the scroll context on error.